### PR TITLE
Update hello-helidon with configmap support

### PR DESCRIPTION
**Pull request by arjav-desai**
VZ-2307

### CODE REVIEW

File: hello-helidon/README.md [hunk 1]
```diff
@@ -3,6 +3,13 @@
 
 This example application provides a simple *Hello World* REST service written with [Helidon](https://helidon.io).
 
+This example application has two endpoint/flavors
```
Comment (kminder) [inline hello-helidon/README.md:6]: Needs to be reviewed by @rosemarymarano

File: hello-helidon/helidon-app-greet-v1/src/main/java/io/helidon/examples/quickstart/mp/Main.java [hunk 2]
```diff
@@ -25,13 +34,31 @@ private Main() {
...
         // load logging configuration
         setupLogging();
 
-        // start the server
-        Server server = startServer();
+        if (isConfig != null) {
+            LOGGER.info("Starting server based on custom configmap configuration");
+            // start the server
+            startConfigServer(args);
+            //System.out.println("http://localhost:" + server.port() + "/config");
```
Comment (kminder) [inline hello-helidon/helidon-app-greet-v1/src/main/java/io/helidon/examples/quickstart/mp/Main.java:46]: Should these commented out lines be removed?

File: hello-helidon/helidon-app-greet-v1/src/main/java/io/helidon/examples/quickstart/mp/Main.java [hunk 3]
```diff
@@ -46,11 +73,38 @@ static Server startServer() {
...
+
+        io.helidon.microprofile.cdi.Main.main(args);
+    }
+
+    /**
+     * Log the change in config
+     * @param config object passed here is the one which is generated onChnage
+     */
+    private static void logConfig(Config config) {
+        LOGGER.info("Loaded config are: " + config.get("config.greeting"));
```
Comment (kminder) [inline hello-helidon/helidon-app-greet-v1/src/main/java/io/helidon/examples/quickstart/mp/Main.java:108]: Wording feels off.  "Configured greeting is:" perhaps?

File: hello-helidon/README.md [hunk 4]
```diff
@@ -3,6 +3,13 @@
...
+- /greet which uses micro-profile properties file as config source
```
Comment (rosemarymarano) [inline hello-helidon/README.md:7]: /greet which -> `/greet`, which (code font and a comma)
source -> source. (period at the end of the sentence)

File: hello-helidon/helidon-app-greet-v2/src/main/java/io/helidon/examples/quickstart/mp/Main.java [hunk 5]
```diff
@@ -25,13 +34,31 @@ private Main() {
...
         // load logging configuration
         setupLogging();
 
-        // start the server
-        Server server = startServer();
+        if (isConfig != null) {
+            LOGGER.info("Starting server based on custom configmap configuration");
+            // start the server
+            startConfigServer(args);
+            //System.out.println("http://localhost:" + server.port() + "/config");
```
Comment (kminder) [inline hello-helidon/helidon-app-greet-v2/src/main/java/io/helidon/examples/quickstart/mp/Main.java:46]: Remove commented out lines?

File: hello-helidon/README.md [hunk 6]
```diff
@@ -3,6 +3,13 @@
...
+- /config which uses k8s configmap as config source
```
Comment (rosemarymarano) [inline hello-helidon/README.md:8]: /config which -> `/config`, which (code font and a comma)
k8s configmap -> Kubernetes ConfigMap
source -> source. (period at the end of the sentence)

File: hello-helidon/README.md [hunk 7]
```diff
@@ -3,6 +3,13 @@
...
+
 ### Install the Example Application
```
Comment (rosemarymarano) [inline hello-helidon/README.md:10]: Install the Example Application -> Install the example application

File: hello-helidon/README.md [hunk 8]
```diff
@@ -3,6 +3,13 @@
...
 
-Detailed instructions for installing Hello World Helidon can be found [here](https://github.com/verrazzano/verrazzano/blob/master/examples/hello-helidon/README.md).
+Detailed instructions for installing Hello World Helidon can be found
```
Comment (rosemarymarano) [inline hello-helidon/README.md:12]: can be found -> can be found at: (colon)

File: hello-helidon/README.md (hunk 1 above)
Comment (arjav-desai) [reply hello-helidon/README.md:6]: Add @rosemarymarano as reviewer.

Comment (rosemarymarano) [inline hello-helidon/README.md:6]: endpoint/flavors -> endpoints: (should be plural, colon at the end)
Not sure why you are using the term flavors. Do you mean that the example application has two variations? Or, something else?

File: hello-helidon/README.md (hunk 4 above)
Comment (rosemarymarano) [inline hello-helidon/README.md:7]: /greet which -> `/greet`, which (code font and comma)
uses micro-profile -> uses a microprofile
as config source -> as a configuration source. (period at the end of a sentence)

File: hello-helidon/README.md (hunk 6 above)
Comment (rosemarymarano) [inline hello-helidon/README.md:8]: /config which -> `/config`, which (code font and comma)
uses k8s configmap -> uses a Kubernetes ConfigMap
as config source -> as a configuration source. (period at the end of a sentence)

File: hello-helidon/README.md (hunk 7 above)
Comment (rosemarymarano) [inline hello-helidon/README.md:10]: Install the Example Application -> Install the example application

File: hello-helidon/README.md (hunk 8 above)
Comment (rosemarymarano) [inline hello-helidon/README.md:12]: can be found -> can be found at: (colon)

File: hello-helidon/README.md [hunk 9]
```diff
@@ -3,6 +3,13 @@
...
+This example application has two endpoint/flavors
+- /greet which uses micro-profile properties file as config source
+- /config which uses k8s configmap as config source
+
 ### Install the Example Application
 
-Detailed instructions for installing Hello World Helidon can be found [here](https://github.com/verrazzano/verrazzano/blob/master/examples/hello-helidon/README.md).
+Detailed instructions for installing Hello World Helidon can be found 
+
+- [/greet](https://github.com/verrazzano/verrazzano/blob/master/examples/hello-helidon/README.md).
```
Comment (rosemarymarano) [inline hello-helidon/README.md:14]: [/greet] -> This points to the Hello World Helidon README and should be described as such.
No period is needed at the end of this bulleted item.

File: hello-helidon/README.md [hunk 10]
```diff
@@ -3,6 +3,13 @@
...
+- [/config](https://github.com/verrazzano/verrazzano/blob/master/examples/helidon-config/README.md).
```
Comment (rosemarymarano) [inline hello-helidon/README.md:15]: [/config] -> This returns a 404 Error.
No period is needed at the end of this bulleted item.

File: hello-helidon/README.md [hunk 11]
```diff
@@ -3,6 +3,13 @@
 
 This example application provides a simple *Hello World* REST service written with [Helidon](https://helidon.io).
```
Comment (rosemarymarano) [inline hello-helidon/README.md:4]: *Hello World* -> Hello World (no need for italics)

**Review by rosemarymarano — COMMENTED** [review]
Lots of edits. Please let me know when you are done incorporating them. Also, if you have any questions on them.

File: hello-helidon/helidon-app-greet-v1/pom.xml [hunk 12]
```diff
@@ -51,8 +51,8 @@
             <optional>true</optional>
         </dependency>
         <dependency>
-            <groupId>javax.activation</groupId>
-            <artifactId>javax.activation-api</artifactId>
+            <groupId>jakarta.activation</groupId>
```
Comment (markxnelson) [inline hello-helidon/helidon-app-greet-v1/pom.xml:54]: this is a new dependency and requires approval

File: hello-helidon/helidon-app-greet-v1/src/main/java/io/helidon/examples/quickstart/mp/ConfigProvider.java [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+package io.helidon.examples.quickstart.mp;
+
+import java.util.function.Supplier;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+/**
```
Comment (markxnelson) [inline hello-helidon/helidon-app-greet-v1/src/main/java/io/helidon/examples/quickstart/mp/ConfigProvider.java:11]: blank line before comment

File: hello-helidon/helidon-app-greet-v1/src/main/java/io/helidon/examples/quickstart/mp/ConfigResource.java [hunk 14]
```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2021, Oracle and/or its affiliates.
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+package io.helidon.examples.quickstart.mp;
```
Comment (markxnelson) [inline hello-helidon/helidon-app-greet-v1/src/main/java/io/helidon/examples/quickstart/mp/ConfigResource.java:3]: new line after copyright header

File: hello-helidon/helidon-app-greet-v1/src/main/java/io/helidon/examples/quickstart/mp/ConfigProvider.java [hunk 15]
```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2021, Oracle and/or its affiliates.
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+package io.helidon.examples.quickstart.mp;
```
Comment (markxnelson) [inline hello-helidon/helidon-app-greet-v1/src/main/java/io/helidon/examples/quickstart/mp/ConfigProvider.java:3]: blank line after license header

**Review by markxnelson — CHANGES_REQUESTED** [review]
comments apply to all occurrences of the same issue, not just the one where they are mentioned

File: hello-helidon/helidon-app-greet-v1/pom.xml (hunk 12 above)
Comment (arjav-desai) [reply hello-helidon/helidon-app-greet-v1/pom.xml:54]: helidon 2.2.1 has brings 1.2.2 of this API based on https://github.com/oracle/helidon/blob/2.2.1/dependencies/pom.xml#L39

File: hello-helidon/README.md (hunk 10 above)
Comment (rosemarymarano) [reply hello-helidon/README.md:15]: That's fine, but please use something more descriptive in the link text.

File: hello-helidon/helidon-app-greet-v1/src/main/java/io/helidon/examples/quickstart/mp/Main.java (hunk 2 above)
Comment (arjav-desai) [reply hello-helidon/helidon-app-greet-v1/src/main/java/io/helidon/examples/quickstart/mp/Main.java]: Removed commented line.

File: hello-helidon/helidon-app-greet-v2/src/main/java/io/helidon/examples/quickstart/mp/Main.java (hunk 5 above)
Comment (arjav-desai) [reply hello-helidon/helidon-app-greet-v2/src/main/java/io/helidon/examples/quickstart/mp/Main.java]: Uncommented it.

File: hello-helidon/helidon-app-greet-v1/src/main/java/io/helidon/examples/quickstart/mp/ConfigProvider.java (hunk 13 above)
Comment (arjav-desai) [reply hello-helidon/helidon-app-greet-v1/src/main/java/io/helidon/examples/quickstart/mp/ConfigProvider.java:13]: Added blank line.

File: hello-helidon/helidon-app-greet-v1/src/main/java/io/helidon/examples/quickstart/mp/ConfigResource.java (hunk 14 above)
Comment (arjav-desai) [reply hello-helidon/helidon-app-greet-v1/src/main/java/io/helidon/examples/quickstart/mp/ConfigResource.java:4]: Added blank line.

File: hello-helidon/helidon-app-greet-v1/src/main/java/io/helidon/examples/quickstart/mp/ConfigProvider.java (hunk 15 above)
Comment (arjav-desai) [reply hello-helidon/helidon-app-greet-v1/src/main/java/io/helidon/examples/quickstart/mp/ConfigProvider.java:4]: Added blank line.

File: hello-helidon/README.md [hunk 16]
```diff
@@ -1,15 +1,11 @@
...
 
-### Install the Example Application
+This example Hello World Helidon application has two endpoints:
 
-Detailed instructions for installing Hello World Helidon can be found 
-
-- [/greet](https://github.com/verrazzano/verrazzano/blob/master/examples/hello-helidon/README.md).
-- [/config](https://github.com/verrazzano/verrazzano/blob/master/examples/helidon-config/README.md).
\ No newline at end of file
+- `/greet`, which uses microprofile properties file as a configuration source. Installation instructions are specified as part of [hello-helidon](https://github.com/verrazzano/verrazzano/blob/master/examples/hello-helidon/README.md) example.
```
Comment (rosemarymarano) [inline hello-helidon/README.md:10]: uses microprofile -> uses a microprofile
Installation instructions are specified as part of -> For installation instructions, see [hello-helidon](https://github.com/verrazzano/verrazzano/blob/master/examples/hello-helidon/README.md).

File: hello-helidon/README.md [hunk 17]
```diff
@@ -1,15 +1,11 @@
...
+- `/config`, which uses a Kubernetes ConfigMap as a configuration source. Installation instructions are specified as part of [helidon-config](https://github.com/verrazzano/verrazzano/blob/master/examples/helidon-config/README.md) example.
```
Comment (rosemarymarano) [inline hello-helidon/README.md:11]: Installation instructions are specified as part of -> For installation instructions, see [helidon-config](https://github.com/verrazzano/verrazzano/blob/master/examples/helidon-config/README.md).

**Review by rosemarymarano — COMMENTED** [review]
Much better. Thank you.

File: hello-helidon/README.md (hunk 16 above)
Comment (arjav-desai) [reply hello-helidon/README.md]: updated.

File: hello-helidon/README.md (hunk 17 above)
Comment (arjav-desai) [reply hello-helidon/README.md]: fixed.

**Review by rosemarymarano — APPROVED** [review]

**Review by markxnelson — APPROVED** [review]

**Pull request merged**